src/app/analyze/stack/stack-details/stack-details.component.ts

```diff
@@ -42,23 +51,190 @@ export class StackDetailsComponent implements OnInit {
+
+    modal = document.createElement('div');
+    modal.classList.add('modal-container');
+
+    let innerModal: Element = document.createElement('div');
+    innerModal.classList.add('modal-inner');
+
+    modal.appendChild(innerModal);
+
+    let head: Element = document.createElement('div');
+    head.innerHTML = modalInformation.header + '<span class="close_icon">x</span>';
+    head.classList.add('modal-head');
+
+    let subject: Element = document.createElement('div');
+    subject.innerHTML = modalInformation.subject;
+    subject.classList.add('modal-subject');
+
+    innerModal.appendChild(head);
+    innerModal.appendChild(subject);
+
+    body.appendChild(modal);
+
+    modal.addEventListener('click', (event) => {
+      let tgt: any = event.target;
+      if (tgt.classList.contains('close_icon')) {
+        this.closeModal();
+      }
+    });
+  }
+
+  closeModal(): void {
+    let body: HTMLElement = document.getElementsByTagName('body')[0];
+
+    let cache: NodeList = document.getElementsByClassName('modal-container');
+
+    if (cache && cache.length > 0) {
+      body.removeChild(cache[0]);
+    }
   }
 
+  /* Modal */
+
+  /* Adding Single Work item */
+  addWorkItem(row: any): void {
+    let workItemData: any = { 'data': { 'attributes': { 'system.state': 'new',
+                              'system.title': 'Sample Test',
+                              'system.description': 'Sample Description to test' },
+                              'relationships':
+                            { 'baseType': { 'data':
+                            { 'id': 'userstory', 'type': 'workitemtypes' } } },
+                            'type': 'workitems', 'id': '55' } };
+
+    workItemData.data.attributes["system.title"] = row.custom.name + ' ' + row.name + ' ' + row.version;
+    let workflow: Observable<any> = this.addWorkFlowService.addWorkFlow(workItemData);
+    workflow.subscribe((data) => {
+      let baseUrl: string = 'http://demo.almighty.io/work-item/list/detail/' + data.data.id;
+      this.openModal({
+        header: 'Response for Work item',
+        subject: 'Successfully created a work item. You can see it here! <a target="_blank" href="' + baseUrl + '">Link</a>'
+      });
+    });
+  }
+  /* Adding Single Work item */
+
+  /* Get Recommendation */
+  getRecommendations(components, recommendation: any): void {
+    this.similarStacks = recommendation.similar_stacks;
+    const analysis: any = this.similarStacks[0].analysis;
+    let missingPackages: Array<any> = analysis.missing_packages;
+    let versionMismatch: Array<any> = analysis.version_mismatch;
+
+    const url: string = this.similarStacks[0].uri;
+    this.recoArray[this.currentIndex]['rows'] = [];
+    this.recoArray[this.currentIndex]['url'] = url;
+    for (var component in components) {
+      this.recoArray[this.currentIndex]['rows'].push({ name: components[component].name, version: components[component].version });
+    }
+    for (let i in missingPackages) {
```

Comment:
The for...in loop will iterate over all enumerable properties of an object.
The for...of syntax is specific to collections, rather than all objects. It will iterate in this manner over the elements of any collection that has a property.

I think this should be "for .. of" not "for .. in"
This also applies to the next loop.